### ISSUE TITLE
Snyk should be a devDependency?

### ISSUE THREAD

**Issue by davidsierradz** [guidesmiths/rascal#98]
Hi! nice project, I installed rascal and noted in my lock file a lot of entries related to snyk, is snyk a requisite to run rascal as a dependency? I don't want to download all of snyk code if its not needed.

Maybe related to this: https://github.com/snyk/snyk/issues/938

**Comment by cressie176** [comment]
Hi, I agree it's somewhat annoying. It's something to do with [sync protect](https://support.snyk.io/hc/en-us/articles/360000911277-Why-has-Snyk-been-added-as-a-dependency-in-my-project-). @UlisesGascon are you able to shed any more light?

**Comment by UlisesGascon** [comment]
@davidsierradz @cressie176  you are right. Let me investigate if we can move the dependency to `devDependency`.

**Comment by bausmeier** [comment]
We've just received a security advisory on some of our repositories for https://github.com/advisories/GHSA-pp7h-53gx-mx7r.

Upon investigation it turned out that the `bl` dependency is a transitive dependency of this module due to its use of `snyk`.

```
Reasons this module exists
   - "rascal#snyk#snyk-docker-plugin#tar-stream" depends on it
   - Hoisted from "rascal#snyk#snyk-docker-plugin#tar-stream#bl"
```

I'm not familiar with Snyk so I'm not sure how to determine if it can be moved to a `devDependency`, but it would be great if we could get an update on this.

Thanks!

**Comment by cressie176** [comment]
That's annoying. It can be used as a dev dependency but limits some of the behaviour. I think it's worth doing, and will make the changes when I get the chance.

**Comment by cressie176** [comment]
Removing snyk in favour of npm audit for now. If I understand correctly, there's nothing to stop users from installing snyk on their projects, and running snyk protect on their own code bases.

**Comment by cressie176** [comment]
Published as rascal@10.2.6
